sample-app/src/components/DropdownSection.tsx

```diff
@@ -1,16 +1,22 @@
-import classNames from "classnames";
 import { useState, useEffect, useRef } from "react";
+import { useHistory } from "react-router";
+
+export interface VerticalLink {
+  label: string,
+  verticalKey: string
+}
 
 export interface Option {
   value: string,
-  display: JSX.Element
+  verticalLinks?: VerticalLink[],
```

Comment:
Ideally, I don't think DropdownSection should know what a VerticalLink is. Is it possible to move `onSelectOption` here so we can have different behavior per option? Then maybe we can get rid of `focusedLinkIndex`. Right now increment/decrementOptionFocus are pretty confusing.

Reply:
if would be a separate function `onFocusOption` that DropdownSection would need, but focusedLinkIndex is needed so DropdownSection can remember which link is in focus instead of the actual option. Alternatively we can have an AutocompleteResult component instead of a render function so it can store its own focus option/link index state and handle its own click/enter event. I will try to see if that would work, otherwise I can refactor increment/decrementOptionFocus into smaller functions

Reply:
seem like we can remove link stuff from dropdownSection unless we want to adopt the same model of inputDropdown + DropdownSection model for AutocompleteOption component to be in charge of navigating the option + links

Reply:
The InputDropdown is currently responsible for determining which section is focused, and each dropdown section is responsible for maintaining the focus within that section. Having the AutocompleteResult maintain it's own focus is doable-but that's a lot of additional complexity. Instead, what do you think about having a `DropdownSectionWithSubOptions` and keeping `DropdownSection` but renaming it to `SimpleDropdownSection`?

Reply:
can't each link be its own option? do we need this idea of sub-options?

Reply:
That could work, I will try it out

Reply:
hmm, ya that might be simpler

Reply:
I can try out the separate section idea. I don't think the link can be its own option? beside the different styling, the link require the corresponding option's value for input update, which is different from the link's label on display, and the click/enter behavior is different. unless we make a `Option` and `OptionLink` and some interface that merges it maybe

Reply:
update: yea having it as its own option works!